ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp

```diff
@@ -137,34 +153,27 @@ void DisplayDashboardView::packSocPercentageReceived(double packSocPercentage)
 {
     ui_.stateOfChargeCapacityWidget().setValue(packSocPercentage);
 
-    if (packSocPercentage > 85)
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(20,180,20)");
-    }
-    else if ((packSocPercentage > 70) && (packSocPercentage <= 85))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(90,190,20)");
-    }
-    else if ((packSocPercentage > 55) && (packSocPercentage <= 70))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(160,200,20)");
-    }
-    else if ((packSocPercentage > 40) && (packSocPercentage <= 55))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,210,20)");
-    }
-    else if ((packSocPercentage > 25) && (packSocPercentage <= 40))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,150,20)");
-    }
-    else if ((packSocPercentage > 10) && (packSocPercentage <= 25))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,90,20)");
-    }
-    else
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,20,20)");
-    }
+    // The rgb values for the progressbar are calculated with the intention of displaying a colour closer to red
+    // for low values and a colour closer to green for higher values. These are calculated using linear equations
+    // with a slope and intercept.
+
+    // Default colour
+    int red = RED_INITIAL;
+    int green = GREEN_INITIAL;
+    int blue = BLUE_INITIAL;
+
+    // Calculated color
+    red += int(RED_SLOPE * packSocPercentage);
+    green += int(GREEN_SLOPE * packSocPercentage);
+
+    QString r = QString::number(red);
+    QString g = QString::number(green);
+    QString b = QString::number(blue);
+
+    QString rgb = "rgb(" + r + "," + g + "," + b + ");";
```

Comment:
Build this with args instead, it's prettier and I think would be less error prone.

http://stackoverflow.com/questions/4784155/how-to-format-a-qstring